docs/source/serving.rst

```diff
@@ -1,12 +1,468 @@
+-------------------------------------------
+
+HuggingFace Transformers provides seamless integration with torchao quantization. The ``TorchAoConfig`` automatically applies torchao's optimized quantization algorithms during model loading. For this example, we'll use `Float8DynamicActivationFloat8WeightConfig` on the Phi-4 mini-instruct model.
+
+.. code-block:: bash
+    pip install git+https://github.com/huggingface/transformers@main
+    pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu126
+    pip install torch
+    pip install accelerate
+
+.. code-block:: python
+
+    import torch
+    from transformers import AutoModelForCausalLM, AutoTokenizer, TorchAoConfig
+
+    model_id = "microsoft/Phi-4-mini-instruct"
+
+    from torchao.quantization import Float8DynamicActivationFloat8WeightConfig, PerRow
+    quant_config = Float8DynamicActivationFloat8WeightConfig(granularity=PerRow())
+    quantization_config = TorchAoConfig(quant_type=quant_config)
+    quantized_model = AutoModelForCausalLM.from_pretrained(model_id, device_map="auto", torch_dtype=torch.bfloat16, quantization_config=quantization_config)
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+    # Push to hub
+    USER_ID = "YOUR_USER_ID"
+    MODEL_NAME = model_id.split("/")[-1]
+    save_to = f"{USER_ID}/{MODEL_NAME}-float8dq"
+    quantized_model.push_to_hub(save_to, safe_serialization=False)
+    tokenizer.push_to_hub(save_to)
+
+    # Manual Testing
+    prompt = "Hey, are you conscious? Can you talk to me?"
+    messages = [
+        {
+            "role": "system",
+            "content": "",
+        },
+        {"role": "user", "content": prompt},
+    ]
+    templated_prompt = tokenizer.apply_chat_template(
+        messages,
+        tokenize=False,
+        add_generation_prompt=True,
+    )
+    print("Prompt:", prompt)
+    print("Templated prompt:", templated_prompt)
+    inputs = tokenizer(
+        templated_prompt,
+        return_tensors="pt",
+    ).to("cuda")
+    generated_ids = quantized_model.generate(**inputs, max_new_tokens=128)
+    output_text = tokenizer.batch_decode(
+        generated_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
+    )
+    print("Response:", output_text[0][len(prompt):])
+
+.. note::
+    For more information on supported quantization and sparsity configurations, see `HF-Torchao Docs <https://huggingface.co/docs/transformers/main/en/quantization/torchao>`_.
+
+Serving and Inference
+--------------------
+
+Serving and Inference with vLLM
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+vLLM automatically leverages torchao's optimized kernels when serving quantized models, providing significant throughput improvements.
+
+First, install vLLM with torchao support:
+
+.. code-block:: bash
+
+    pip install vllm --pre --extra-index-url https://wheels.vllm.ai/nightly
+    pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu126
+
+.. code-block:: bash
```

Comment:
I think we need to add some text here. E.g. we need to explain we're serving with the quantized checkpoint we pushed to HF hub above. Also would be good to clarify that "float8dq" stands for "dynamic quant"